File: implementations/go/query/query_test.go
package query

import "testing"

func TestQueryEventTypesIncludesAllEntries(t *testing.T) {
	expected := []string{
		"query.requested",
		"query.response",
		"query.rejected",
		"query.error",
	}
	for _, typ := range expected {
		if !EventTypes[typ] {
			t.Errorf("missing: %s", typ)
		}
	}
	if len(EventTypes) != len(expected) {
		t.Errorf("size mismatch: got %d, want %d", len(EventTypes), len(expected))
	}
}

func TestIsEventTypePositives(t *testing.T) {
	tests := []string{
		"query.requested",
		"query.response",
		"query.rejected",
		"query.error",
	}
	for _, typ := range tests {
		if !IsEventType(typ) {
			t.Errorf("expected true for %s", typ)
		}
	}
}

func TestIsEventTypeNegatives(t *testing.T) {
	tests := []string{
		"task.submitted",
		"session.opened",
		"command.requested",
		"",
	}
	for _, typ := range tests {
		if IsEventType(typ) {
			t.Errorf("expected false for %s", typ)
		}
	}
}

File: implementations/typescript/packages/query/test/query.test.js
import assert from "node:assert/strict";
import test from "node:test";
import { QUERY_EVENT_TYPES, isQueryEventType } from "../src/index.js";

test("query event types includes all 4 registry entries", () => {
  const expected = [
    "query.requested",
    "query.response",
    "query.rejected",
    "query.error"
  ];
  for (const type of expected) {
    assert.equal(QUERY_EVENT_TYPES.has(type), true, `missing: ${type}`);
  }
  assert.equal(QUERY_EVENT_TYPES.size, expected.length);
});

test("isQueryEventType positives", () => {
  assert.equal(isQueryEventType("query.requested"), true);
  assert.equal(isQueryEventType("query.response"), true);
  assert.equal(isQueryEventType("query.rejected"), true);
  assert.equal(isQueryEventType("query.error"), true);
});

test("isQueryEventType negatives", () => {
  assert.equal(isQueryEventType("task.submitted"), false);
  assert.equal(isQueryEventType("session.opened"), false);
  assert.equal(isQueryEventType("command.requested"), false);
  assert.equal(isQueryEventType(""), false);
});
